### PR TITLE
SONARPY-1522 Update which Python versions are supported

### PR DESCRIPTION
**Pull request by Seppli11**
[SONARPY-1522](https://sonarsource.atlassian.net/browse/SONARPY-1522)

Adds stubs for Python 3.12 and 3.13 (however, typeshed and other stubs weren't updated. Therefore, new and updated APIs don't behave correctly yet).
Furthermore, Python 3.6 and 3.7 were removed, as they are EOL.

[SONARPY-1522]: https://sonarsource.atlassian.net/browse/SONARPY-1522?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by guillaume-dequenne-sonarsource — CHANGES_REQUESTED** [review]
Looks good!
I have some questions on the changed tests, but no specific challenge against them, let's discuss it before making any changes.
I think the missing coverage on the defensive throw is okay.

File: python-frontend/src/test/java/org/sonar/python/semantic/v2/typeshed/TypeShedDescriptorsProviderTest.java [hunk 1]
```diff
@@ -111,13 +111,13 @@ void cacheTest() {
   @Test
   void stdlibDescriptors() {
```
[the remark this answers is not in the document]
Comment (Seppli11) [reply python-frontend/src/test/java/org/sonar/python/semantic/v2/typeshed/TypeShedDescriptorsProviderTest.java]: `math.acos` stopped being ambiguous. To keep the test mostly the same, I've changed it to use `os.path.realpath`, as it is still ambiguous.

To test the `annotatedReturnTypeName` again, I'll add a test to check the `annotatedReturnTypeName` of `math.acos`

File: python-frontend/src/test/java/org/sonar/python/semantic/v2/typeshed/TypeShedDescriptorsProviderTest.java [hunk 2]
```diff
@@ -68,9 +68,9 @@ void builtin312DescriptorsTest() {
   void builtinFloatDisambiguation() {
     var provider = typeshedDescriptorsProvider();
     var builtinDescriptors = provider.builtinDescriptors();
-    var floatDescriptor = builtinDescriptors.get("float");
```
[the remark this answers is not in the document]
Comment (Seppli11) [reply python-frontend/src/test/java/org/sonar/python/semantic/v2/typeshed/TypeShedDescriptorsProviderTest.java]: the `__new__` of `float` isn't ambiguous anymore and therefore the test fails. `int.to_bytes` still is

[the remark this answers is not in the document]
Comment (Seppli11) [reply python-frontend/src/test/java/org/sonar/python/semantic/v2/typeshed/TypeShedDescriptorsProviderTest.java]: As part of the review-fixes I changed the name of the test to reflect better that is isn't just about floats

**Review by guillaume-dequenne-sonarsource — APPROVED** [review]
LGTM.
Note that PR analysis test failed due to reasons unrelated to this PR.
This other [PR](urlhttps://github.com/SonarSource/sonar-python/pull/2207) fixes it. Please make sure to rebase when the PR is merged, so that we can have a green CI

**Comment by Seppli11** [comment]
I've regenerated the stubs multiple times, and no differences were reported by git. So, it seems that the stubs generation is stable, at least on linux.

**Pull request merged**